src/models/Model.h

```diff
@@ -26,6 +30,9 @@ class Model {
         Logic const & logic, SymRef sr,
         std::string const & formalArgDefaultPrefix); // Return a string that is not equal to the argument
 
+    static std::unique_ptr<Model> extend(Model const & base, std::span<std::pair<PTRef, PTRef>> extension);
```

Comment:
Why `static`?

Reply:
.. and we may even consider a non-const (or `&&`) variant.

Reply:
You mean `static` compared to member function?
Just because that this is an operation _on_ the object rather than an operation _of_ the object.
If you prefer member function, I could live with that.

Variants on what? The model?

Reply:
In my view, it is an operation of the object. It makes a copy of itself and adds extra variables.
I think that static functions are often used as factory functions where the object is not available yet. Personally, it just confused me why it is `static` and I needed to take a look whether there is an extra reason fo that.

I think that the case where the object is also non-const is useful too, if we do not need the previous model.
For example:
```
Model & Model::extend(PTRef var, PTRef val);
```
However, to avoid name clash, it should be named differently than the variant `std::unique_ptr<Model> Model::extend(PTRef var, PTRef val) const;`.
I propose to get inspired by [algorithm](https://en.cppreference.com/w/cpp/algorithm) library and name the const variant `extendCopy`, or `copyAndExtend`. Or, to name the non-const variant `inplaceExtend`.

Reply:
I changed it to be a member function, though I disagree with it.
The reason is related to why we cannot have `inplaceExtend`: the models were designed to be immutable, once you get a model you can only use it to evaluate terms, but you cannot change it. The members related to the evaluation of the base terms are `const` and I think they should stay that way.

That's why I also see the extension as operation on the model, not of the model. The model itself is immutable (from logical perspective).

Reply:
I think that in the future, it should be still perfectly fine to at least allow to move-construct the new extended model.